### PR TITLE
Make it easier to solve warnings/errors referring to project settings

### PR DESCRIPTION
**Pull request by RedMser**
- Always include full path to project setting in warning and error messages, for easy copy-pasting.
- When filtering for project settings, include advanced settings even in basic mode.

image

In this example, Enable Input is an advanced setting, but shows up anyway because we are searching for it.

### CODE REVIEW

**Comment by RedMser** [comment]
> When I search with a keyword (e.g. "input"), I want to see all settings that relate to that search.

That's actually a fair point. Changed it so a non-empty filter now shows advanced settings results as well.

File: drivers/gles3/storage/material_storage.cpp [hunk 1]
```diff
@@ -1358,7 +1358,7 @@ MaterialStorage::MaterialStorage() {
 	global_shader_uniforms.buffer_size = MAX(4096, (int)GLOBAL_GET("rendering/limits/global_shader_variables/buffer_size"));
 	if (global_shader_uniforms.buffer_size > uint32_t(Config::get_singleton()->max_uniform_buffer_size)) {
 		global_shader_uniforms.buffer_size = uint32_t(Config::get_singleton()->max_uniform_buffer_size);
-		WARN_PRINT("Project setting: rendering/limits/global_shader_variables/buffer_size exceeds maximum uniform buffer size of: " + itos(Config::get_singleton()->max_uniform_buffer_size));
+		WARN_PRINT("Project setting 'rendering/limits/global_shader_variables/buffer_size' exceeds maximum uniform buffer size of: " + itos(Config::get_singleton()->max_uniform_buffer_size));
```
Comment (Calinou) [inline drivers/gles3/storage/material_storage.cpp:1361]: Use double quotes for consistency with GDScript style guide and other engine messages:

```suggestion
		WARN_PRINT("Project setting \"rendering/limits/global_shader_variables/buffer_size\" exceeds maximum uniform buffer size of: " + itos(Config::get_singleton()->max_uniform_buffer_size));
```

File: servers/audio/audio_stream.cpp [hunk 2]
```diff
@@ -370,7 +370,7 @@ void AudioStreamPlaybackMicrophone::start(double p_from_pos) {
 	}
 
 	if (!GLOBAL_GET("audio/driver/enable_input")) {
-		WARN_PRINT("Need to enable Project settings > Audio > Enable Audio Input option to use capturing.");
+		WARN_PRINT("Need to enable project setting 'audio/driver/enable_input' to use capturing.");
```
Comment (Calinou) [inline servers/audio/audio_stream.cpp:373]:
```suggestion
		WARN_PRINT("You must enable the project setting \"audio/driver/enable_input\" to use audio capture.");
```

File: servers/rendering/renderer_rd/environment/sky.cpp [hunk 3]
```diff
@@ -1791,7 +1791,7 @@ void SkyRD::update_dirty_skys() {
 			if (sky->mode == RS::SKY_MODE_REALTIME) {
 				layers = 8;
 				if (roughness_layers != 8) {
-					WARN_PRINT("When using REALTIME skies, roughness_layers should be set to 8 in the project settings for best quality reflections");
+					WARN_PRINT("When using REALTIME skies, 'rendering/reflections/sky_reflections/roughness_layers' should be set to 8 in the project settings for best quality reflections");
```
Comment (Calinou) [inline servers/rendering/renderer_rd/environment/sky.cpp:1794]:
```suggestion
					WARN_PRINT("When using the Real-Time sky update mode, \"rendering/reflections/sky_reflections/roughness_layers\" should be set to 8 in the project settings for best quality reflections.");
```

**Review by clayjohn — APPROVED** [review]
Implementation looks fine to me. 

@akien-mga Can you confirm that you are happy with the changes?

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**